cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1133,19 +1135,23 @@ private boolean isPodUpToDate(StatefulSet ss, Pod pod) {
             return ssGeneration == podGeneration;
         }
 
-        private boolean isPodCaCertUpToDate(Pod pod) {
-            final int caCertGeneration = getClusterCaCertGeneration();
+        private boolean isPodCaCertUpToDate(Pod pod, Ca ca) {
+            final int caCertGeneration = getCaCertGeneration(ca);
+            String podAnnotation = ca instanceof ClientsCa ?
+                    Ca.ANNO_STRIMZI_IO_CLIENTS_CA_CERT_GENERATION :
+                    Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION;
             final int podCaCertGeneration =
-                    Integer.parseInt(pod.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION));
+                    Integer.parseInt(pod.getMetadata().getAnnotations().get(podAnnotation));
             return caCertGeneration == podCaCertGeneration;
         }
 
         private boolean isPodToRestart(StatefulSet ss, Pod pod, boolean isAncillaryCmChange, Ca... cas) {
             boolean isPodUpToDate = isPodUpToDate(ss, pod);
-            boolean isPodCaCertUpToDate = isPodCaCertUpToDate(pod);
+            boolean isPodCaCertUpToDate = true;
             boolean isCaCertsChanged = false;
             for (Ca ca: cas) {
                 isCaCertsChanged = isCaCertsChanged || ca.certRenewed() || ca.certsRemoved();
+                isPodCaCertUpToDate = isPodCaCertUpToDate && isPodCaCertUpToDate(pod, ca);
```

Comment:
```suggestion
                isPodCaCertUpToDate &= isPodCaCertUpToDate(pod, ca);
```

Reply:
@tombentley agree and we can also do following for the line before that one:

```
isCaCertsChanged |= ca.certRenewed() || ca.certsRemoved();
```